### ISSUE TITLE
Feature Request: Move IR Tx to a hardware PWM capable pin

### ISSUE THREAD

**Issue by normanr** [AnaviTechnology/anavi-infrared#6]
If you spin another hardware revision it would be nice if the IR Tx could be moved to a PWM capable pin. Pin 12 (BCM GPIO 18) is one of the few ports that support hardware PWM, so just switching the RX and TX pins would do, and would be pretty easy as their traces run very close to each other on the board, and would be easy to manually reroute.

**Comment by leon-anavi** [comment]
Hi @normanr,

This is a great idea for an improvement! Thank you for the feedback. The only thing that worries me is that it may create confusion among users of the current and the potential new hardware revision.

Eventually, it may be worth doing this on a new HAT (with the full HAT size) with some additional features.

Thanks,
Leon

**Comment by normanr** [comment]
Yes, I considered the confusion. I think if the board is clearly marked with Rev 2.0, and it's documented, then maybe it would be okay. Another alternative would be to use pin 32/33/35 (the other PWM pins), maybe with a jumper (or solder bridge?) to select between pin 11 and the PWM capable pin.

While I've swapped pin 11 and 12 with some rework, I haven't yet switched to the pwm-ir-tx driver, so I'm still using the gpio-ir-tx driver which just bit-bangs the output. So I have yet to measure the performance difference that swapping the pins enables.

fwiw one of the reasons why I love the ANAVI Infrared pHAT was that it was the same size as the Pi Zero. About the only thing lacking I've found is a 5V output, as I'm looking to connect a PMS5003 to it, and it requires 5V power with 3V3 UART (currently connected via a FTDI TTL-232-3V3 cable).

**Comment by leon-anavi** [comment]
Hi @normanr,

Using a jumper is a wonderful idea because it will allow swapping the GPIOs without any soldering. Adding 5V next to the 3.3V to the UART header is also a great idea and easy to do. Both definitely makes sense for rev 2.0.

As of the moment I am not familiar with the `pwm-ir-tx` drive. Could you please provide feedback if you have a chance to give it a try in the future?

Thanks again for the feedback and sharing these great ideas for improvements.

Thanks,
Leon
